### PR TITLE
CD: auto deploy codes on the master branch to the online demo

### LINKED ISSUE #920: bug: 修改upstream权重为0熔断后，修改rout会出现问题
### 修改upstream
image
### 修改rout
image

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

Fix #920 

should deploy code in master branch to the online demo

### CODE REVIEW

File: .github/workflows/deploy-with-docker.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ on:
   push:
     branches:
       - master
+  pull_request:
+    branches:
+      - master
```
Comment (juzhiyuan) [inline .github/workflows/deploy-with-docker.yml:9]: emmm No

File: docs/deploy-with-docker.md [hunk 2]
```diff
@@ -38,6 +38,10 @@ $ docker build -t apisix-dashboard:$tag .
 
 # For users in mainland China, the `ENABLE_PROXY` parameter can be provided to speed up module downloads.
 $ docker build -t apisix-dashboard:$tag . --build-arg ENABLE_PROXY=true
+
+# If you want to use the latest code to build, you can specify the `APISIX_DASHBOARD_VERSION` parameter to `master`. 
+# This parameter can also be specified as a version number, such as `v2.0`.
+$ docker build -t apisix-dashboard:$tag . --build-arg ENABLE_PROXY=true --build-arg APISIX_DASHBOARD_VERSION=master
```
Comment (juzhiyuan) [inline docs/deploy-with-docker.md:44]: `ENABLE_PROXY` is not needed;

File: docs/deploy-with-docker.md [hunk 3]
```diff
@@ -38,6 +38,10 @@ $ docker build -t apisix-dashboard:$tag .
 
 # For users in mainland China, the `ENABLE_PROXY` parameter can be provided to speed up module downloads.
 $ docker build -t apisix-dashboard:$tag . --build-arg ENABLE_PROXY=true
+
+# If you want to use the latest code to build, you can specify the `APISIX_DASHBOARD_VERSION` parameter to `master`. 
+# This parameter can also be specified as a version number, such as `v2.0`.
```
Comment (juzhiyuan) [inline docs/deploy-with-docker.md:43]: xxx version string, actually the release branch's name

File: docs/deploy-with-docker.md [hunk 4]
```diff
@@ -38,6 +38,10 @@ $ docker build -t apisix-dashboard:$tag .
 
 # For users in mainland China, the `ENABLE_PROXY` parameter can be provided to speed up module downloads.
 $ docker build -t apisix-dashboard:$tag . --build-arg ENABLE_PROXY=true
+
+# If you want to use the latest code to build, you can specify the `APISIX_DASHBOARD_VERSION` parameter to `master`.
```
Comment (juzhiyuan) [inline docs/deploy-with-docker.md:42]: codes

File: docs/deploy-with-docker.zh-CN.md [hunk 5]
```diff
@@ -38,6 +38,9 @@ $ docker build -t apisix-dashboard:$tag .
 
 # 对于中国大陆的用户，可启用 `ENABLE_PROXY` 参数加快模块下载速度。
 $ docker build -t apisix-dashboard:$tag . --build-arg ENABLE_PROXY=true
+
+# 如果需要使用最新代码构建，可启用 `APISIX_DASHBOARD_VERSION` 参数指定为 `master` ，此参数也可以指定为其他版本号，如 `v2.0` 。
```
Comment (juzhiyuan) [inline docs/deploy-with-docker.zh-CN.md:42]: ditto

**Review by juzhiyuan — CHANGES_REQUESTED** [review]

File: docs/deploy-with-docker.md (hunk 4 above)
Comment (membphis) [reply docs/deploy-with-docker.md:42]: @juzhiyuan do you mean to change `code` to `codes`?

Comment (juzhiyuan) [reply docs/deploy-with-docker.md:42]: Yes, it's a grammar issue.

File: docs/deploy-with-docker.zh-CN.md (hunk 5 above)
Comment (nic-chen) [reply docs/deploy-with-docker.zh-CN.md]: fixed.

File: docs/deploy-with-docker.md (hunk 4 above)
Comment (nic-chen) [reply docs/deploy-with-docker.md]: fixed.

File: docs/deploy-with-docker.md (hunk 3 above)
Comment (nic-chen) [reply docs/deploy-with-docker.md]: fixed.

File: docs/deploy-with-docker.md (hunk 2 above)
Comment (nic-chen) [reply docs/deploy-with-docker.md]: fixed.

File: .github/workflows/deploy-with-docker.yml (hunk 1 above)
Comment (nic-chen) [reply .github/workflows/deploy-with-docker.yml]: removed.

File: docs/deploy-with-docker.md [hunk 6]
```diff
@@ -38,6 +38,10 @@ $ docker build -t apisix-dashboard:$tag .
 
 # For users in mainland China, the `ENABLE_PROXY` parameter can be provided to speed up module downloads.
 $ docker build -t apisix-dashboard:$tag . --build-arg ENABLE_PROXY=true
+
+# If you want to use the latest codes to build, you can specify the `APISIX_DASHBOARD_VERSION` parameter to `master`. 
+# This parameter can also be specified as branch name of a specific version, such as `v2.0`.
+$ docker build -t apisix-dashboard:$tag . --build-arg APISIX_DASHBOARD_VERSION=master
```
Comment (membphis) [inline docs/deploy-with-docker.md:44]: what is the default value of `APISIX_DASHBOARD_VERSION`?

Comment (nic-chen) [reply docs/deploy-with-docker.md:44]: now it's v2.0. maybe master is better ?

Comment (membphis) [reply docs/deploy-with-docker.md:44]: I think it should be 2.1 now. 

APISIX has release 2.1 today ^_^

**Comment by membphis** [comment]
@nic-chen CI failed, please take a look

File: docs/deploy-with-docker.md (hunk 6 above)
Comment (juzhiyuan) [reply docs/deploy-with-docker.md:44]: no, use 2.1 is because the dashboard has released the 2.1.

**Comment by nic-chen** [comment]
> @nic-chen CI failed, please take a look

OK now, front end e2e test unstable this time.

**Review by membphis — APPROVED** [review]

File: docs/deploy-with-docker.md (hunk 6 above)
Comment (membphis) [reply docs/deploy-with-docker.md:44]: @juzhiyuan you are right

File: docs/deploy-with-docker.zh-CN.md [hunk 7]
```diff
@@ -38,6 +38,9 @@ $ docker build -t apisix-dashboard:$tag .
 
 # 对于中国大陆的用户，可启用 `ENABLE_PROXY` 参数加快模块下载速度。
 $ docker build -t apisix-dashboard:$tag . --build-arg ENABLE_PROXY=true
+
+# 如果需要使用最新代码构建，可启用 `APISIX_DASHBOARD_VERSION` 参数指定为 `master` ，此参数也可以指定为其他版本的分支名，如 `v2.0` 。
```
Comment (juzhiyuan) [inline docs/deploy-with-docker.zh-CN.md:42]: Users may not know what's it, need more clearer or simpler description.

**Review by juzhiyuan — APPROVED** [review]
Merge first for I need to trigger CD.

**Pull request merged**